### PR TITLE
ARUHA-2637 Use transactions while creating event type

### PR DESCRIPTION
**Pull request by antban**
ARUHA-2637 Use transactions while creating event type

Also, As a part of the same problem - do not reset cache during transaction, run event type cache update only after successful transaction commit, as updating cache in the middle of transaction only leads to bugs.

### CODE REVIEW

**Comment by antban** [comment]
:+1:

File: src/main/java/org/zalando/nakadi/service/EventTypeService.java [hunk 1]
```diff
@@ -176,18 +177,33 @@ public void create(final EventTypeBase eventType, final boolean checkAuth)
...
             }
-            throw e;
+            if (null != createdEventType.get()) {
+                try {
+                    eventTypeRepository.removeEventType(createdEventType.get().getName());
+                } catch (RuntimeException ex2) {
+                    LOG.warn("Failed to rollback event type creation (which is highly expected)", ex2);
+                }
+            }
+            throw ex;
```
Comment (adyach) [inline src/main/java/org/zalando/nakadi/service/EventTypeService.java]: maybe you want to wrap it in `InternalNakadiException` and give a good description

**Comment by adyach** [comment]
Not all calls to event type update are handled with cache update:
`TimelineCleanupJob.markTimelineDeleted`
`TimelineService.updateTimeLineForRepartition`
etc

**Comment by antban** [comment]
@adyach omg, thanks for catching it!

**Comment by antban** [comment]
:+1:

**Comment by adyach** [comment]
👍

**Pull request merged**